tests/integration/ha/test_backups.py

```diff
@@ -310,40 +316,14 @@ async def test_restore_cluster_after_app_destroyed(ops_test: OpsTest) -> None:
         timeout=1400,
         idle_period=IDLE_PERIOD,
     )
-
-    units = await get_application_unit_ids_ips(ops_test, app=app)
-    leader_id = await get_leader_unit_id(ops_test, app)
-
-    action = await run_action(ops_test, leader_id, "restore", params={"backup-id": 1})
-    logger.info(f"restore output: {action}")
-    assert action.status == "completed"
-
-    # index document
-    doc_id = TEST_BACKUP_DOC_ID
-    # check that the doc can be retrieved from any node
-    logger.info("Test backup index: searching")
-    for u_id, u_ip in units.items():
-        docs = await search(
-            ops_test,
-            app,
-            u_ip,
-            TEST_BACKUP_INDEX,
-            query={"query": {"term": {"_id": doc_id}}},
-            preference="_only_local",
-        )
-        # Validate the index and document are present
-        assert len(docs) == 1
-        assert docs[0]["_source"] == default_doc(TEST_BACKUP_INDEX, doc_id)
+    # This is the same check as the previous restore action.
+    # Call the method again
+    await test_restore_cluster(ops_test)
```

Comment:
It's a bit awkward one test calling the other...

I'm trying to decide if that's a good trick or not...

Don't we wanna have a helper function with the body of this test, and that would be called both in an individual test, and in here?

Reply:
Yes, I am adding one, so we can also keep the c_writes.

Reply:
Now, we have ContinuousWrites in the Backup test (01). Then, for restore, DR (disaster recovery) and relation removed/readded scenarios, we start the ContinuousWrites *after* the restoration of the cluster.

That is because our goal is to validate the cluster is functional post-restore.